File: 2022/elixir/test/day02_test.exs
defmodule Day02Test do
  use ExUnit.Case

  @example_input """
  A Y
  B X
  C Z
  """

  test "part one", do: assert(Day02.part_one(@example_input) == 15)
  test "part two", do: assert(Day02.part_two(@example_input) == 12)
end

File: 2024/python/day02_test.py
"""
Advent of Code 2024, Day 2: Red-Nosed Reports.

See: https://adventofcode.com/2024/day/2
"""

import io

import pytest

from day02 import part_one, part_two

EXAMPLE = """\
7 6 4 2 1
1 2 7 8 9
9 7 6 2 1
1 3 2 4 5
8 6 4 4 1
1 3 6 7 9
"""


@pytest.fixture()
def example() -> str:
    return EXAMPLE


def test_part_one(example: str) -> None:
    """
    Test that checks if the solution for part one works on the example input.
    """
    assert part_one(io.StringIO(example)) == 2


def test_part_two(example: str) -> None:
    """
    Test that checks if the solution for part two works on the example input.
    """
    assert part_two(io.StringIO(example)) == 4
